test/units/cli/test_galaxy.py

```diff
@@ -444,3 +459,228 @@ def test_template_ignore_similar_folder(self):
+        assert os.listdir(os.path.join(collection_skeleton, empty_dir)) == []
+
+    # Test files that don't end with .j2 were not templated
+    doc_file = os.path.join(collection_skeleton, 'docs', 'My Collection.md')
+    assert os.path.exists(doc_file)
+    with open(doc_file, 'r') as f:
+        doc_contents = f.read()
+    assert doc_contents.strip() == 'Welcome to my test collection doc for {{ namespace }}.'
+
+    # Test files that end with .j2 but are in the templates directory were not templated
+    for template_dir in ['playbooks/templates', 'playbooks/templates/subfolder',
+                         'roles/common/templates', 'roles/common/templates/subfolder']:
+        test_conf_j2 = os.path.join(collection_skeleton, template_dir, 'test.conf.j2')
+        assert os.path.exists(test_conf_j2)
+
+        with open(test_conf_j2, 'r') as f:
+            contents = f.read()
+        expected_contents = '[defaults]\ntest_key = {{ test_variable }}'
+
+        assert expected_contents == contents.strip()
+
+
+@pytest.fixture()
+def collection_build(collection_skeleton):
+    output_dir = tempfile.mkdtemp()
+    if not os.path.isdir(output_dir):
+        os.makedirs(output_dir)
+
+    # Because we call GalaxyCLI in collection_skeleton we need to reset the singleton back to None so it uses the new
+    # args, we reset the original args once it is done.
+    orig_cli_args = co.GlobalCLIArgs._Singleton__instance
+    try:
+        co.GlobalCLIArgs._Singleton__instance = None
+        galaxy_args = ['ansible-galaxy', 'collection', 'build', collection_skeleton, '--output-path', output_dir]
+        gc = GalaxyCLI(args=galaxy_args)
+        gc.run()
+
+        yield output_dir
+    finally:
+        co.GlobalCLIArgs._Singleton__instance = orig_cli_args
+
+    if os.path.isdir(output_dir):
+        shutil.rmtree(output_dir)
+
+
+def test_invalid_skeleton_path():
+    expected = "- the skeleton path '/fake/path' does not exist, cannot init collection"
+
+    gc = GalaxyCLI(args=['ansible-galaxy', 'collection', 'init', 'my.collection', '--collection-skeleton',
+                         '/fake/path'])
+    with pytest.raises(AnsibleError, match=expected):
+        gc.run()
+
+
+@pytest.mark.parametrize("name", ["invalid", "hypen-ns.collection", "ns.hyphen-collection", "ns.collection.weird"])
+def test_invalid_collection_name(name):
+    expected = "Invalid collection name, must be in the format <namespace>.<collection>"
+
+    gc = GalaxyCLI(args=['ansible-galaxy', 'collection', 'init', name])
+    with pytest.raises(AnsibleError, match=expected):
+        gc.run()
+
+
+@pytest.mark.parametrize('collection_skeleton', [
+    ('ansible_test.build_collection', None),
+], indirect=True)
+def test_collection_build(collection_build):
+    tar_path = os.path.join(collection_build, 'ansible_test-build_collection-1.0.0.tar.gz')
+    assert os.path.exists(tar_path)
+    assert tarfile.is_tarfile(tar_path)
+
+    tar = tarfile.open(tar_path)
```

Comment:
I thought we agreed that it'd be cleaner with a CM. Weren't you going to implement it? Did I miss it somewhere?

Reply:
I did https://github.com/ansible/ansible/pull/57106/commits/648a2520707a5fd56a1add386576fbf1d6fac4fa#diff-2de3c8cc3e7fae814d02b3058f0c8f9eR124, this is just for a test that I can just import and use it with. Will update on the next commit.

Reply:
Ah, I was looking for it in test files and missed that :)